org.ekstep.sunbirdcommonheader-1.8/editor/headerApp.js

```diff
@@ -770,6 +783,172 @@ angular.module('org.ekstep.sunbirdcommonheader:app', ["Scope.safeApply", "yaru22
+        $scope.$safeApply(function(){
+            ecEditor.jQuery('#errorLockContentModal').modal({
+                inverted: true,
+                closable: false,
+                onVisible: function(){
+                    ecEditor.jQuery('#errorLockContentModal').mouseover(function(){
+                        if($scope.isIdle){
+                            ecEditor.jQuery('#errorLockContentModal').modal('hide');
+                        }
+                    });
+                },
+                onDeny: function() {
+                    $scope.closeEditor();
+                },
+                onApprove: function() {
+                    $scope.contentDataChanged();
+                    $scope.validateContentLock();
+                }
+            }).modal('show');
+        });
+     }
+   }
+   $scope.refreshContent = function() {
+    location.reload();
+   }
+
+    $scope.validateContentLock = function () {
+        //console.log("called ", $scope.contentLockListener);
+        var lastSyncTime = $scope.lastContentLockSyncTime.getTime();
+        var currentTime = (new Date()).getTime();
+        var timeDiff = currentTime - lastSyncTime;
+        $scope.idleTimer += $scope.contentLockRefershInterval;
+        // if screen is active(not idle)then refresh the lock regularly 
+        if($scope.dataChanged === true || $scope.previewMode === true){
+            try {
+             $scope.refreshContentLock();
+            } catch(e) {
+             console.log("err ",e)
+            }
+            $scope.dataChanged = false;
+            $scope.idleTimer = 0;
+            $scope.lastContentLockSyncTime = new Date();
+            return;
+        }
+        // if lock expires then show resume/close message
+        if(Math.floor(timeDiff/1000) >=  $scope.contentLockExpiresIn) {
+            try {                
+                $scope.showStatusPopup('SESSION_TIMEOUT');
+                $scope.contentLockExpired = true;
+            } catch(e) {
+             console.log("err ",e)
+            }
+            $scope.idleTimer = 0;
+            $scope.lastContentLockSyncTime = new Date();
+            return;
+        }
+
+         // if user is idle and lock not expired then show idle screen
+        if($scope.idleTimer >=  $scope.contentLockIdleTimeOut && $scope.contentLockExpired === false) {
+            // save content if any changes before showing idle screen
+            if($scope.disableSaveBtn === false){
+                $scope.saveContent(function(err,res){});
+            }
+            $scope.idleTimer = 0;            
+            $scope.showStatusPopup('IDLE_TIMEOUT');
+            return;
+        }
+    }
+
+     $scope.setContentLockListener = function (event) {
+        if($scope.contentLockListener){
+          $scope.removeContentLockListener()
+        }
+        if($scope.lockObj && $scope.lockObj.lockKey){
+            //convert to seconds
+            $scope.contentLockExpiresIn = $scope.lockObj.expiresIn*60;
```

Comment:
We should handle mandatory fields by showing proper toast in the starting of the editor launch itself. So that it will be easy to identify the defects in the starting before going through detailed analysis.